src/live-updates.ts

```diff
@@ -0,0 +1,50 @@
+export type Entity = Record<string, unknown>;
+export type Argument = Entity | Entity[];
+export type SubscribeCallback = (data: Argument) => void;
+
+/**
+ * LiveUpdates for the Contentful Live Preview mode
+ * receives the updated Entity from the Editor and merges them together with the provided data
+ */
+export class ContentfulLiveUpdates {
+  private subscriptions: { id: number; data: Argument; locale: string; cb: SubscribeCallback }[] =
+    [];
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  private mergeGraphQL(initial: Argument, locale: string, updated: Entity): Argument {
```

Comment:
fyi, we will need the content type here as well for entries. I can try to receive that from the host app on my part.